src/bin/pg_rewind/t/107_empty_conf.pl

```diff
@@ -0,0 +1,50 @@
+use strict;
+use warnings;
+use TestLib;
+use Test::More tests => 3;
+
+use FindBin;
+use lib $FindBin::RealBin;
+use File::Copy;
+
+use RewindTest;
+
+sub run_test
+{
+	my $test_mode     = shift;
+
+	RewindTest::setup_cluster($test_mode);
+	RewindTest::start_master();
+	RewindTest::create_standby($test_mode);
+	RewindTest::promote_standby(1);
+
+	my $master_pgdata = $node_master->data_dir();
+	my $tmp_folder    = TestLib::tempdir;
+
+	# pg_rewind should handle empty (or even removed) postgres.conf
+	# gp_dbid is taken from internal.auto.conf
+	copy(
+		"$master_pgdata/postgresql.conf",
+		"$tmp_folder/master-postgresql-full.conf.tmp");
+
+	open my $file, '>', "$master_pgdata/postgresql.conf";
+	print $file "";
```

Comment:
What happens without this print? The previous line is enough to create the empty file.